src/Agent.Sdk/Util/LoggedSecretMasker.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.TeamFoundation.DistributedTask.Logging;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Agent.Sdk.Util
+{
+    /// <summary>
+    /// Extended secret masker service, that allows to log origins of secrets
+    /// </summary>
+    public class LoggedSecretMasker : ISecretMasker
+    {
+        private ISecretMasker _secretMasker;
+        private ITraceWriter _trace;
+
+        private void Trace(string msg)
+        {
+            this._trace?.Info($"[DEBUG INFO]{msg}");
```

Comment:
Do we really need this prefix? I just used this in original draft changes for debug purpose only

Reply:
Yeah, it could be removed